adapters.go

```diff
@@ -67,15 +74,19 @@ func (a *whisperAdapter) RetrievePublicMessages(chatID string) ([]*protocol.Mess
 		return nil, err
 	}
 
+	logger := a.logger.With(zap.String("site", "RetrievePublicMessages"))
+
 	decodedMessages := make([]*protocol.Message, 0, len(messages))
 	for _, item := range messages {
 		shhMessage := whisper.ToWhisperMessage(item)
 
-		log.Printf("[whisperAdapter::RetrievePublicMessages] received a public message: %#x", shhMessage.Hash)
+		hlogger := logger.With(zap.String("hash", fmt.Sprintf("%#x", shhMessage.Hash)))
```

Comment:
I wonder how would `zap.Binary()` render this in stdout.

Reply:
An example for 
```
zap.Binary("their-identity-key", ecrypto.FromECDSAPub(theirIdentityKey))
```
would be:
```
2019-07-17T17:34:11.508+0200    INFO    encryption/encryptor.go:599     built a message {"encryptor": {"site": "EncryptPayload", "their-identity-key": "BJJcYMg1wdxWVZ82yQMncGdf68fMnrr5TbH8rbx1+Q6Vo55jgzoVeDzEFeULOthfPHkTKsdx/Bc2sxUo139tclk=", "installation-ids": ["1"]}}
```